aea/registries/base.py

```diff
@@ -225,292 +237,165 @@ def _add_protocol(self, protocol_directory: Path):
-        """
-        Set up the handlers in the registry.
-
-        :return: None
-        """
-        for skill_id_to_handler_dict in self._handlers.values():
-            for handler in skill_id_to_handler_dict.values():
-                handler.setup()
-
-    def teardown(self) -> None:
-        """
-        Teardown the registry.
-
-        :return: None
-        """
-        for skill_id_to_handler_dict in self._handlers.values():
-            for skill_id, handler in skill_id_to_handler_dict.items():
-                try:
-                    handler.teardown()
-                except Exception as e:
-                    logger.warning(
-                        "An error occurred while tearing down handler {}/{}: {}".format(
-                            skill_id, type(handler).__name__, str(e)
-                        )
-                    )
-        self._handlers = {}
-
-
-class BehaviourRegistry(Registry):
-    """This class implements the behaviour registry."""
+            )
+        self._items.setdefault(skill_id, {})[item_name] = item
 
-    def __init__(self) -> None:
+    def unregister(self, item_id: Tuple[SkillId, str]) -> None:
         """
-        Instantiate the registry.
+        Unregister a item.
 
+        :param item_id: a pair (skill id, item name).
         :return: None
         """
-        self._behaviours = {}  # type: Dict[SkillId, List[Behaviour]]
 
-    def register(self, ids: Tuple[None, SkillId], behaviours: List[Behaviour]) -> None:
-        """
-        Register a behaviour.
+        skill_id = item_id[0]
+        item_name = item_id[1]
+        name_to_handlers = self._items.get(skill_id, {})
+        name_to_handlers.pop(item_name, None)
 
-        :param ids: the skill id.
-        :param behaviours: the behaviours of the skill.
-        :return: None
-        """
-        skill_id = ids[1]
-        if skill_id in self._behaviours.keys():
-            logger.warning(
-                "Behaviours already registered with skill id '{}'".format(skill_id)
-            )
-        self._behaviours.setdefault(skill_id, []).extend(behaviours)
+        if len(name_to_handlers) == 0:
+            self._items.pop(skill_id, None)
 
-    def unregister(self, skill_id: SkillId) -> None:
-        """
-        Unregister a behaviour.
-
-        :param skill_id: the skill id.
-        :return: None
+    def fetch(self, item_id: Tuple[SkillId, str]) -> Optional[SkillComponentType]:
         """
-        self._behaviours.pop(skill_id, None)
+        Return a item.
 
-    def fetch(self, skill_id: SkillId) -> Optional[List[Behaviour]]:
+        :return: the list of items
         """
-        Return a behaviour.
+        skill_id = item_id[0]
+        item_name = item_id[1]
+        return self._items.get(skill_id, {}).get(item_name, None)
 
-        :return: the list of behaviours
-        """
-        return self._behaviours.get(skill_id, None)
+    def fetch_by_skill(self, skill_id: SkillId) -> List[Item]:
```

Comment:
this is a method not defined in the `Registry` interface, because that makes sense only for `SkillComponentRegistry`. It seems it is pretty useful across the project.

Reply:
Ok